src/ai/ObjectDetectorYOLO.ts

```diff
@@ -0,0 +1,90 @@
+        'microwave', 'oven', 'toaster', 'sink', 'refrigerator', 'book', 'clock', 'vase', 'scissors', 'teddy bear',
+        'hair drier', 'toothbrush'];
+    // private static predictions: DetectedObject[];
+    public static async loadModel(callback?: () => any) {
+        const path = '/yolov5/model.json';
+        await tf.loadGraphModel(path).then((model  ) => {
+            ObjectDetectorYolov5.model = model;
+            ObjectDetectorYolov5.AIModel= AIModel.OBJECT_DETECTION_YOLOv5;
+            store.dispatch(updateObjectDetectorStatus(true));
+            store.dispatch(updateActiveLabelType(LabelType.RECT));
+            const activeLabelType: LabelType = LabelsSelector.getActiveLabelType();
+            if (activeLabelType === LabelType.RECT) {
+                AIObjectDetectionActions.detectRectsForActiveImage();
+            }
+            if (callback) {
+                callback();
+            }
+        }).catch((error) => {
+            store.dispatch(updateObjectDetectorStatus(false));
+            throw new Error(error as string);
+        });
+    }
+
+    public static imgToTensor(img) {
+        const imgTensor = tf.browser.fromPixels(img);
+        const originHeight = img.height;
+        const originWidth = img.width;
+        const inputTensor = tf.image
+            .resizeBilinear(imgTensor, [ObjectDetectorYolov5.height, ObjectDetectorYolov5.width])
+            .div(255.0)
+            .expandDims(0);
+        return [inputTensor, originHeight, originWidth];
+    }
+
+    public static async predict(image: HTMLImageElement, callback?: (predictions: DetectedObject[]) => any) {
+        if (!ObjectDetectorYolov5.model) return;
+        tf.engine().startScope();
+        const [input, originHeight, originWidth] = ObjectDetectorYolov5.imgToTensor(image);
+        const predictions: DetectedObject[] = [];
+        const results = await ObjectDetectorYolov5.model.executeAsync(input);
+        const boxes = await results[0].dataSync();
+        const scores = await results[1].dataSync();
+        const classes = await results[2].dataSync();
+        const validDetections = await results[3].dataSync();
+        for (let i = 0; i < validDetections; i++) {
+            let [x1, y1, x2, y2] = boxes.slice(i * 4, (i + 1) * 4);
+            x1 *= originWidth;
+            x2 *= originWidth;
+            y1 *= originHeight;
+            y2 *= originHeight;
+            const width = x2 - x1;
+            const height = y2 - y1;
+            const className = ObjectDetectorYolov5.names[classes[i]];
+            const score = scores[i];
+            predictions.push({
+                        bbox: [x1, y1, width, height],
+                        class: className,
+                        score: score.toFixed(2)
+                    });
+        }
+        tf.dispose(results);
+        tf.engine().endScope();
+        if (callback) {
+            callback(predictions);
+        }
+    }
+}
```

Comment:
Requires refactoring, adding a confidence threshold a user can select would be nice. I was training the model, but in the first iterations the confidence of identifying my object was low. Unfortunately i was not able to change the confidence coefficient.